nion/swift/ExportDialog.py

```diff
@@ -221,56 +226,155 @@ def cancel(self) -> bool:
-        main_page = u.create_column(width_row, height_row, spacing=12, margin=12)
-
-        self.ui_view = main_page
+        self.ui_view = u.create_column(
+            u.create_row(
+                u.create_label(text=_("Width:"), width=80),
+                u.create_line_edit(
+                    text="@binding(width_model.value)",
+                    on_text_edited="width_value_updated",
+                    background_color="white",
+                ),
+                spacing=12
+            ),
+            u.create_row(
+                u.create_label(text=_("Height:"), width=80),
+                u.create_line_edit(
+                    text="@binding(height_model.value)",
+                    on_text_edited="height_value_updated",
+                    background_color="white",
+                ),
+                spacing=12
+            ),
+            u.create_row(
+                u.create_label(text=_("Units:"), width=80),
+                u.create_combo_box(
+                    items=[_("Inches"), _("Centimeters"), _("Pixels")],
+                    current_index="@binding(unit_model.value)",
+                    on_current_index_changed="unit_changed"
+                ),
+                spacing=12
+            ),
+            spacing=12,
+            margin=12
+        )
+
+    def width_value_updated(self, widget: UserInterface.LineEditWidget, text: str) -> None:
+        try:
+            new_width = float(text)
+            new_height = new_width / self.aspect_ratio
+            if self.unit_model == UnitType.PIXELS:
+                new_height = round(new_height)
+            self.height_model.value = str(new_height)
+        except ValueError:
+            pass  # Handle the case where the input is not a valid number
+
+    def height_value_updated(self, widget: UserInterface.LineEditWidget, text: str) -> None:
+        try:
+            new_height = float(text)
+            new_width = new_height * self.aspect_ratio
+            if self.unit_model == UnitType.PIXELS:
+                new_width = round(new_width)
+            self.width_model.value = str(new_width)
+        except ValueError:
+            pass  # Handle the case where the input is not a valid number
+
+    def unit_changed(self, widget: UserInterface.ComboBoxWidget, current_index: int) -> None:
+        new_unit_type = UnitType(current_index)
+        width_text = self.width_model.value
+        height_text = self.height_model.value
+
+        self.size = self.convert_to_pixels(
+            Geometry.FloatSize(height=float(height_text or 0),
+                               width=float(width_text or 0)),
+            self.previous_unit_index
+        )
+        newshape = self.convert_from_pixels(self.size, new_unit_type)
+
+        if new_unit_type == UnitType.PIXELS:
+            new_width:float = round(newshape.width)
+            new_height:float = round(newshape.height)
```

Comment:
These ":float" parts should be unnecessary and should be removed.

Also, if they were required for some reason, please a space after the ":":
`new_width: float = ...`